In [2]:
import pandas as pd

In [3]:
file = [
    'land_cover/landcover1.csv',
    'land_cover/landcover2.csv',
    'land_cover/landcover3.csv',
    'land_cover/landcover4.csv',
    'land_cover/landcover5.csv',
]

In [4]:
# merging two csv files
df = pd.concat(
    map(pd.read_csv, file), ignore_index=True)
df.head()

discrete_classification  discrete_classification-proba  bare-coverfraction  \
0                      200                            NaN                 NaN   
1                       40                           96.0                 0.0   
2                       40                           88.0                 0.0   
3                      126                           32.0                 8.0   
4                       40                           97.0                 0.0   

   urban-coverfraction  crops-coverfraction  grass-coverfraction  \
0                  NaN                  NaN                  NaN   
1                  0.0                100.0                  0.0   
2                  0.0                 81.0                  0.0   
3                  0.0                 19.0                 29.0   
4                  0.0                 96.0                  0.0   

   moss-coverfraction  water-permanent-coverfraction  \
0                 NaN                            NaN   
1                 0.0                            0.0   
2                 0.0                            0.0   
3                 0.0                            0.0   
4                 0.0                            0.0   

   water-seasonal-coverfraction  shrub-coverfraction  snow-coverfraction  \
0                           NaN                  NaN                 NaN   
1                           0.0                  0.0                 0.0   
2                          95.0                  0.0                 0.0   
3                           0.0                 10.0                 0.0   
4                           2.0                  4.0                 0.0   

   tree-coverfraction  forest_type  data-density-indicator  system:index  \
0                 NaN          NaN                     NaN             0   
1                 0.0          NaN                    82.0             1   
2                19.0          0.0                    82.0             2   
3                34.0          0.0                    79.0             3   
4                 0.0          NaN                    81.0             4   

    name  
0  node0  
1  node1  
2  node2  
3  node3  
4  node4

In [5]:
len(df)

19824

In [6]:
# merge this column to the regression data csv
df2 = pd.read_csv('regression_data.csv')
df2['land_cover'] = df['discrete_classification']
df2.head()

latitude   longitude  region   state  postcode  \
0  6.402858  100.098777  Kangar  Perlis      1000   
1  6.365790  100.188445  Kangar  Perlis      1000   
2  6.360280  100.212395  Kangar  Perlis      1000   
3  6.557335  100.314105  Kangar  Perlis      1000   
4  6.395771  100.230483  Kangar  Perlis      1000   

                                       geometry  flood_risk     distance  \
0  POINT (11142944.906933343 714251.0946146951)       False  4245.737601   
1  POINT (11152926.669637946 710098.9597010214)       False  4532.533863   
2  POINT (11155592.793706343 709481.7137045604)       False  5266.664356   
3  POINT (11166915.132510776 731557.9727265495)       False  8652.136771   
4  POINT (11157606.329523865 713457.2321468537)       False  3753.265280   

      kawasan_banjir  latitude_kb  longitude_kb  kedalaman_banjir  month_kb  \
0    Kampung Nelayan       6.4040      100.1369               0.6       4.0   
1     Kampung Sentua       6.4051      100.1981               0.2      10.0   
2     Kampung Sentua       6.4051      100.1981               0.2      10.0   
3  Taman Beseri Jaya       6.5590      100.2364               0.6       5.0   
4     Kampung Sentua       6.4051      100.1981               0.2      10.0   

   altitude     slope  min_distance_river  land_cover  
0         0  0.000000         2866.669879         200  
1         4  2.075604         2520.893227          40  
2         5  2.075599          394.344896          40  
3        59  0.933516         2065.659934         126  
4         5  0.927410         2072.528557          40

In [7]:
df2.to_csv('regression_data.csv', index=False)

# What does the land cover code mean? 
The link to the imagery(https://developers.google.com/earth-engine/datasets/catalog/COPERNICUS_Landcover_100m_Proba-V-C3_Global#bands)

Classification:
- 0: Unknown. No or not enough satellite data available.
- 20: Shrubs. Woody perennial plants with persistent and woody stems and without any defined main stem being less than 5 m tall. The shrub foliage can be either evergreen or deciduous.
- 30: Herbaceous vegetation. Plants without persistent stem or shoots above ground and lacking definite firm structure. Tree and shrub cover is less than 10 %.
- 40: Cultivated and managed vegetation / agriculture. Lands covered with temporary crops followed by harvest and a bare soil period (e.g., single and multiple cropping systems). Note that perennial woody crops will be classified as the appropriate forest or shrub land cover type.
- 50: Urban / built up. Land covered by buildings and other man-made structures.
- 60: Bare / sparse vegetation. Lands with exposed soil, sand, or rocks and never has more than 10 % vegetated cover during any time of the year.
- 70: Snow and ice. Lands under snow or ice cover throughout the year.
- 80: Permanent water bodies. Lakes, reservoirs, and rivers. Can be either fresh or salt-water bodies.
- 90: Herbaceous wetland. Lands with a permanent mixture of water and herbaceous or woody vegetation. The vegetation can be present in either salt, brackish, or fresh water.
- 100: Moss and lichen.
- 111: Closed forest, evergreen needle leaf. Tree canopy >70 %, almost all needle leaf trees remain green all year. Canopy is never without green foliage.
- 112: Closed forest, evergreen broad leaf. Tree canopy >70 %, almost all broadleaf trees remain green year round. Canopy is never without green foliage.
- 113: Closed forest, deciduous needle leaf. Tree canopy >70 %, consists of seasonal needle leaf tree communities with an annual cycle of leaf-on and leaf-off periods.
- 114: Closed forest, deciduous broad leaf. Tree canopy >70 %, consists of seasonal broadleaf tree communities with an annual cycle of leaf-on and leaf-off periods.
- 115: Closed forest, mixed.
- 116: Closed forest, not matching any of the other definitions.
- 121: Open forest, evergreen needle leaf. Top layer- trees 15-70 % and second layer- mixed of shrubs and grassland, almost all needle leaf trees remain green all year. Canopy is never without green foliage.
- 122: Open forest, evergreen broad leaf. Top layer- trees 15-70 % and second layer- mixed of shrubs and grassland, almost all broadleaf trees remain green year round. Canopy is never without green foliage.
- 123: Open forest, deciduous needle leaf. Top layer- trees 15-70 % and second layer- mixed of shrubs and grassland, consists of seasonal needle leaf tree communities with an annual cycle of leaf-on and leaf-off periods.
- 124: Open forest, deciduous broad leaf. Top layer- trees 15-70 % and second layer- mixed of shrubs and grassland, consists of seasonal broadleaf tree communities with an annual cycle of leaf-on and leaf-off periods.
- 125: Open forest, mixed.
- 126: Open forest, not matching any of the other definitions.
- 200: Oceans, seas. Can be either fresh or salt-water bodies.

